File: week6/윤여준/위장.cpp
#include <bits/stdc++.h>
using namespace std;

map<string, int> m;

int solution(vector<vector<string>> clothes) {
    int answer = 1;
    
    for(auto vs : clothes) m[vs[1]]++;
    
    for(auto psi : m) answer *= psi.second + 1;
    
    return answer - 1;
}

File: week6/박현국/위장.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        Map<String, Integer> data = new HashMap<>();

        for (String[] cloth : clothes) {
            String item = cloth[0];
            String type = cloth[1];

            int val = data.getOrDefault(type, 0);
            data.put(type, val + 1);
        }

        for (String key : data.keySet()) {
            answer *= data.get(key) + 1;
        }

        return answer - 1;
    }
}